File: energetica/schemas/map.py
"""Schemas for API routes for the map."""

from pydantic import BaseModel


class HexTileOut(BaseModel):
    id: int
    q: int
    r: int
    solar: float
    wind: float
    hydro: float
    coal: float
    gas: float
    uranium: float
    climate_risk: float
    player_id: int | None


class SettleRequest(BaseModel):
    region_id: int


class SettleResponse(BaseModel):
    player_id: int

File: frontend/src/types/map.ts
/**
 * Type definitions for the map. These types correspond to the backend schemas
 * in energetica/schemas/map.py
 */

// TODO: I feel like this should be retrievable from the generated API types.

interface HexTileOut {
    id: number;
    q: number;
    r: number;
    solar: number;
    wind: number;
    hydro: number;
    coal: number;
    gas: number;
    uranium: number;
    climate_risk: number;
    player_id: number | null;
}
